CleverPush/CleverPush/Source/CleverPush.m

```diff
@@ -1462,21 +1464,65 @@ + (void)handleJSONNSURLResponse:(NSURLResponse*) response data:(NSData*) data er
     }
 }
 
++ (void)addSubscriptionTags:(NSArray*)tagIds {
+    [self addSubscriptionTags:tagIds callback:nil];
+}
+
++ (void)removeSubscriptionTags:(NSArray*)tagIds {
+    [self removeSubscriptionTags:tagIds callback:nil];
+}
+
++ (void)addSubscriptionTags:(NSArray*)tagIds callback:(void(^)(NSArray *))callback {
+    dispatch_group_t group = dispatch_group_create();
+    for (NSString* tagId in tagIds) {
+        dispatch_group_enter(group);
+        [self addSubscriptionTag:tagId callback:^(NSString *tagId) {
+            dispatch_group_leave(group);
+        }];
+    }
+    dispatch_group_notify(group, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+        if (callback) {
+            callback([self getSubscriptionTags]);
+        }
+    });
+}
+
++ (void)removeSubscriptionTags:(NSArray*)tagIds callback:(void(^)(NSArray *))callback{
+    dispatch_group_t group = dispatch_group_create();
```

Comment:
What happens if I call this method in the main thread?
Will it block?

Reply:
Nope it will not block.
I have checked this scenario by putting this method on main thread.